<launch>
    <arg name="namespace"     default="drone0" />
    <group ns="$(arg namespace)">
        <arg name="launch_prefix" default="" />

        <arg name="node_namespace" default="apriltag_ros_continuous_node" />

        <arg name="camera_name" default="sensor_measurement/camera" />

        <arg name="camera_frame" default="drone0_image" />

        <arg name="image_topic" default="image_rect" />

        <!-- Set parameters -->
        <rosparam command="load" file="$(find tello_sdk)/config/settings.yaml" ns="$(arg node_namespace)" />

        <rosparam command="load" file="$(find tello_sdk)/config/tags.yaml" ns="$(arg node_namespace)" />

        <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="$(arg node_namespace)" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)" >
            <remap from="image_rect" to="$(arg camera_name)/$(arg image_topic)" />
            <remap from="camera_info" to="$(arg camera_name)/camera_info" />

            <param name="camera_frame" type="str" value="$(arg camera_frame)" />
            <param name="publish_tag_detections_image" type="bool" value="true" />
        </node>
    </group>
</launch>